src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,462 @@
+                error: 'Please select an organism',
+            });
+            return null;
+        }
+
+        if (topNumFeatures.length == 0) {
+            this.setState({
+                error:
+                    'No gene list selected. At least one gene list is required.',
+            });
+            return null;
+        }
+        if (topNumFeatures.length == 0) {
+            this.setState({
+                error:
+                    'No gene list selected. At least one gene list is required.',
+            });
+            return null;
+        }
+
+        const sortedFeatureMetricTable = this.featureMetricTable.sort(
+            (a, b) => b[selectedSortBy] - a[selectedSortBy]
+        );
+        const topFeatureQuery = this.buildFeatureQuery(
+            topNumFeatures,
+            sortedFeatureMetricTable
+        );
+        const organism =
+            gProfilerToken !== null ? gProfilerToken : selectedOrganism;
+        const gProfilerLink = this.buildGProfilerLink(
+            organism,
+            topFeatureQuery
+        );
+        if (gProfilerLink.length > 8000) {
+            this.setState({
+                error:
+                    'Too many genes in total. Try to select a combination of gene lists with fewer genes.',
+            });
+            return null;
+        }
+        return gProfilerLink;
+    };
+
+    setAvailableOrganisms(availableOrganisms) {
+        const _availableOrganisms = availableOrganisms
+            .map((availableOrganism, idx: number) => {
+                return {
+                    key: idx + 1,
+                    text: availableOrganism['display_name'],
+                    value: availableOrganism['id'],
+                };
+            })
+            .sort((a, b) => (a['text'] < b['text'] ? -1 : 1));
+        this.setState({ availableOrganisms: _availableOrganisms });
+    }
+
+    async fetchAvailableOrganisms() {
+        let response = null;
+        try {
+            response = await fetch(
+                'https://biit.cs.ut.ee/gprofiler/api/util/organisms_list/'
+            );
+            return response.json();
+        } catch (err) {
+            this.setState({
+                error: `Unable to fetch list of organisms: ${err}`,
+            });
+        }
+        return response;
+    }
+
+    async componentDidMount() {
+        const availableOrganisms = await this.fetchAvailableOrganisms();
+        if (availableOrganisms !== null)
+            this.setAvailableOrganisms(availableOrganisms);
+    }
+
+    render() {
+        const {
+            showModal,
+            selectedOrganism,
+            selectedSortBy,
+            gProfilerToken,
+        } = this.state;
+
+        return (
+            <>
+                <Modal
+                    as={Form}
+                    trigger={
+                        <Button
+                            color='orange'
+                            onClick={this.handleOpenModal}
```

Comment:
Just `setState` directly here.

Reply:
This is a bad practice

Reply:
"bad practice" is not useful reasoning. Why? What's bad about it? Under what circumstances?

Reply:
Using `() => setState({...})`, you will declare a new function on each component rendering

Reply:
OK so the implied concern is that we'll overload the GC? This model is performance sensitive? This isn't very convincing. I might be swayed by a "readability" argument